### PR TITLE
[aot] Emit the weak field indexes table using the MONO_AOT_TABLE code, so it works with separate aot data fiels/bitcode.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/6948

### CODE REVIEW

**Comment by vargaz** [comment]
This will need a backport to 2017-10/2018-02.

File: mono/mini/aot-runtime.c [hunk 1]
```diff
@@ -2201,6 +2203,8 @@ if (container_assm_name && !container_amodule) {
 		amodule->extra_method_info_offsets = (guint32 *)info->extra_method_info_offsets;
 		amodule->got_info_offsets = info->got_info_offsets;
 		amodule->llvm_got_info_offsets = info->llvm_got_info_offsets;
+		amodule->llvm_got_info_offsets = info->llvm_got_info_offsets;
+		amodule->weak_field_indexes = info->weak_field_indexes;
```
Comment (lewurm) [inline mono/mini/aot-runtime.c]: copy pasta?

there is `amodule->llvm_got_info_offsets = info->llvm_got_info_offsets;` twice.

File: mono/mini/aot-compiler.c [hunk 2]
```diff
@@ -9890,27 +9890,29 @@ emit_image_table (MonoAotCompile *acfg)
...
+	encode_int (g_hash_table_size (indexes), p, &p);
 	g_hash_table_iter_init (&iter, indexes);
 	while (g_hash_table_iter_next (&iter, &key, &value)) {
 		guint32 index = GPOINTER_TO_UINT (key);
-		emit_int32 (acfg, index);
+		encode_int (index, p, &p);
 	}
+	g_assert (p - buf <= buf_size);
+
+	emit_aot_data (acfg, MONO_AOT_TABLE_WEAK_FIELD_INDEXES, "weak_field_indexes", buf, p - buf);
```
Comment (lewurm) [inline mono/mini/aot-compiler.c]: missing `g_free (buf)`

**Comment by marek-safar** [comment]
@monojenkins backport 2017-12

**Comment by marek-safar** [comment]
@monojenkins backport 2018-02

**Comment by brendanzagaeski** [comment]
> This will need a backport to 2017-10

As a quick correction for bookkeeping, `2017-10` in the [earlier comment](https://github.com/mono/mono/pull/7007#issuecomment-365403495) was a small typo for `2017-12`. The 2 backports listed above for `2017-12` and `2018-02` are all that's needed.

(I tested locally to confirm that I was able to hit the build error using a Xamarin.iOS CI package based on mono `2017-12`. Then I downgraded to the current Xamarin.iOS 11.8 preview version (based on mono `2017-10`), and rebuilt the solution _without_ error.)

**Pull request merged**